src/config_brain_stats.js

```diff
@@ -1,5 +1,14 @@
 'use strict';
 
+brain.stats.addRole = function(role) {
+  let userName = Memory.username || _.find(Game.spawns, 'owner').owner;
+  Memory.stats = Memory.stats || {};
+  Memory.stats[userName].roles = Memory.stats[userName].roles || {};
+  let roleStat = Memory.stats[userName].roles[role];
+  let previousAmount = roleStat ? roleStat : 0;
+  Memory.stats[userName].roles[role] = previousAmount + 1;
+};
+
```

Comment:
https://github.com/TooAngel/screeps/pull/203/files#diff-b0aa1decfe0b4b8e62777279273e7a61R123
feel free to copy paste it like anything in my WIP PRs if you like^^